plugins/outputs/sensu/sensu.go

```
@@ -0,0 +1,468 @@
+
+# [outputs.sensu-go.tags]
+#   source = "telegraf"
+
+# [outputs.sensu-go.metrics]
+#   handlers = ["elasticsearch","timescaledb"]
+`
+
+// Description provides a description of the plugin
+func (s *Sensu) Description() string {
+	return "Send aggregate metrics to Sensu Monitor"
+}
+
+// SampleConfig provides a sample configuration for the plugin
+func (s *Sensu) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Sensu) SetSerializer(serializer serializers.Serializer) {
+	s.serializer = serializer
+}
+
+func (s *Sensu) createClient(ctx context.Context) (*http.Client, error) {
+	tlsCfg, err := s.ClientConfig.TLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+			Proxy:           http.ProxyFromEnvironment,
+		},
+		Timeout: s.Timeout.Duration,
+	}
+
+	return client, nil
+}
+
+func (s *Sensu) Connect() error {
+	if s.Timeout.Duration == 0 {
+		s.Timeout.Duration = defaultClientTimeout
+	}
+
+	ctx := context.Background()
+	client, err := s.createClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	s.client = client
+
+	return nil
+}
+
+func (s *Sensu) Close() error {
+	return nil
+}
+
+func (s *Sensu) Write(metrics []telegraf.Metric) error {
+	var points []*OutputMetric
+	for _, metric := range metrics {
+		// Add tags from config to each metric point
+		var tagList []*OutputTag
+		for name, value := range s.Tags {
+			metric.AddTag(name, value)
+		}
+		for _, tagSet := range metric.TagList() {
+			tag := &OutputTag{
+				Name:  tagSet.Key,
+				Value: tagSet.Value,
+			}
+			tagList = append(tagList, tag)
+		}
+
+		// Get all valid numeric values, convert to float64
+		for _, fieldSet := range metric.FieldList() {
+			key := fieldSet.Key
+			// don't need err since math.NaN is returned with err
+			value, _ := getFloat(fieldSet.Value)
+			// JSON does not support these special values
+			if math.IsNaN(value) || math.IsInf(value, 0) {
+				continue
```

How about replace at least the `Inf`s with `+-math.MaxFloat64`? Anyway, you should output at least some debug message notifying the user that you dropped a metric. If we silently drop metrics he/she has no way to find out what happened to their data...

---

Fixed.